### PR TITLE
Restrict deleting internal admin and system roles through backend

### PR DESCRIPTION
**Pull request by ZiyamSanthosh**
With this PR, A validation has been added to avoid deleting both `Internal/admin` and `Internal/system` roles.

Related issue:
- https://github.com/wso2/product-is/issues/18045

### CODE REVIEW

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java [hunk 1]
```diff
@@ -591,6 +593,27 @@ public void deleteRole(String roleId, String tenantDomain) throws IdentityRoleMa
...
+     * @param roleId           Role ID.
+     * @param tenantDomain     Tenant domain.
+     * @param userRealm        User realm.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    private boolean isInternalAdminOrSystemRole(String roleId, String tenantDomain, UserRealm userRealm)
+            throws IdentityRoleManagementException, UserStoreException {
+
+        RoleBasicInfo roleBasicInfo = getRoleBasicInfoById(roleId, tenantDomain);
+        if (ORGANIZATION.equalsIgnoreCase(roleBasicInfo.getAudience())) {
```
Comment (chamathns) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java:608]: add a comment here

Comment (ZiyamSanthosh) [reply components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java:610]: Addressed with https://github.com/wso2/carbon-identity-framework/pull/5245/commits/3e0ed9e8ac826218041ba7b60d18ce960f33663f

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java [hunk 2]
```diff
@@ -591,6 +594,50 @@ public void deleteRole(String roleId, String tenantDomain) throws IdentityRoleMa
...
+     * @param roleId           Role ID.
+     * @param tenantDomain     Tenant domain.
+     * @param userRealm        User realm.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    private boolean isInternalAdminOrSystemRole(String roleId, String tenantDomain, UserRealm userRealm)
+            throws IdentityRoleManagementException, UserStoreException {
+
+        RoleBasicInfo roleBasicInfo = getRoleBasicInfoById(roleId, tenantDomain);
+        // There won't be multiple internal/admin and internal/system roles in the same organization
```
Comment (SujanSanjula96) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java:609]:
```suggestion
        // There won't be multiple internal/admin and internal/system roles in the same organization with organization audience.
```

**Review by SujanSanjula96 — APPROVED** [review]

**Comment by ZiyamSanthosh** [comment]
> PR builder completed Link: https://github.com/wso2/product-is/actions/runs/7042967702 Status: **failure**

The PR builder has failed due to this.
https://github.com/wso2/product-is/pull/18117#issuecomment-1831238662

**Review by chamathns — APPROVED** [review]

**Pull request merged**